test/Pool.test.js

```diff
@@ -1315,6 +1315,196 @@ describe('Pool', function () {
+        }).then(function() {
+          assert.strictEqual(workerCount, 1);
+        });
+    });
+  });
+
+
+  it('should not terminate worker if abort listener is defined inline worker with Timeout', function () {
+    var workerCount = 0;
+    var pool = createPool({
+      onCreateWorker: () => {
+        workerCount += 1;
+      },
+      maxWorkers: 1,
+    });
+    function asyncTimeout() {
+      var me = this;
+      return new Promise(function () {
+        let timeout = setTimeout(function() {
+            resolve();
+        }, 5000); 
+        me.worker.addAbortListener(function () {
+          return new Promise(function (resolve) {
+            clearTimeout(timeout);
+            resolve();
+          });
+        });
+      });
+    }
+    function add(a, b) { }
+    return pool.exec(asyncTimeout, [],  {
+    })
+    .timeout(200)
+    .catch(function(err) {
+      assert(err instanceof Promise.TimeoutError);
+      assert(pool.stats().activeTasks = 0);
+    }).always(function () {
+      return pool.exec(add, [1, 2]).then(function () {
+        assert.strictEqual(workerCount, 1);
+      }); 
+    });
+  });
+
+  it('should not terminate worker if abort listener is defined inline worker with Cancellation', function () {
+    var workerCount = 0;
+    var pool = createPool({
+      onCreateWorker: () => {
+        workerCount += 1;
+      },
+      maxWorkers: 1,
+    });
+    function asyncTimeout() {
+      var me = this;
+      return new Promise(function () {
+        let timeout = setTimeout(function() {
+            resolve();
+        }, 5000); 
+        me.worker.addAbortListener(function () {
+          return new Promise(function (resolve) {
+            clearTimeout(timeout);
+            resolve();
+          });
+        });
+      });
+    }
+    function add(a, b) { }
+    const task = pool.exec(asyncTimeout, [],  {
+    })
+    return new Promise(function(resolve) {
+      setTimeout(function() {
+        resolve();
+      }, 50);
+    }).then(function() {
+      return task
+      .cancel()
+      .catch(function(err) {
+        assert(err instanceof Promise.TimeoutError);
+        assert(pool.stats().activeTasks = 0);
+      }).always(function () {
+        return pool.exec(add, [1, 2]).then(function () {
+          assert.strictEqual(workerCount, 1);
+        }); 
+      });
+    });
+
+  });
+
+  it('should invoke timeout for abort handler if timeout period is reached with Timeout', function () {
+    var workerCount = 0;
+    var pool = createPool(__dirname + '/workers/cleanup-abort.js', {
+      maxWorkers: 1,
+      onCreateWorker: function() {
+        workerCount += 1;
+      }
+    });
+  
+    return pool.exec('asyncAbortHandlerNeverResolves', [])
+    .timeout(1000)
+    .catch(function (err) {
+      assert(err instanceof Promise.TimeoutError);
+
+      var stats = pool.stats();
+      assert(stats.busyWorkers === 1);
+    }).always(function() {
+      return pool.exec(add, [1, 2]).then(function() {
```

Comment:
Instead, shouldn't we check here that `stats.totalWorkers` is zero?

Reply:
Wouldn't the worker count be `1` here since we have finished the task and the `add` task did not cause the worker to terminate? here are the stats from the pool after `add` finishes

```
activeTasks = 0
busyWorkers = 0
idleWorkers = 1
pendingTasks = 0
totalWorkers = 1
```

This makes sense to me as the total number of workers is `1` and the total number of idle workers is `1` so the worker can pick up the next task queued. This is the behavior I think we want correct?

Reply:
Sorry, yes I mean testing `stats.totalWorkers === 1`, not zero.